### PR TITLE
feat: use search manager (TEMP)

### PR DESCRIPTION
**Pull request by rpenido**

### CODE REVIEW

**Comment by yusuf-musleh** [comment]
@rpenido I've created a small [PR](https://github.com/open-craft/frontend-app-course-authoring/pull/45) on this PR that makes some style changes to the filters. I think it small enough to merge it directly to include it as part of this PR.

File: src/library-authoring/components/LibraryComponents.tsx [hunk 1]
```diff
@@ -65,28 +63,7 @@ const LibraryComponents = ({
...
-      // State without loads; show content.
-      resultShowLoading = false;
-      resultShowContent = true;
-    }
-    return {
-      showLoading: resultShowLoading,
-      showContent: resultShowContent,
-    };
-  }, [isFetching, isFetchingNextPage]);
+  const showLoading = isFetching || isFetchingNextPage;
```
Comment (rpenido) [inline src/library-authoring/components/LibraryComponents.tsx:66]: Removing the loading state when we already have data to make the transition smooth.
With loading | Without loading 
:---        |    :----   
components-with-loading | components-without-loading

File: src/search-manager/FilterByTags.tsx
Comment (yusuf-musleh) [inline src/search-manager/FilterByTags.tsx:1]: The labels in the Tags filter dropdown are not aligned:

Screenshot 2024-07-05 at 2 41 58 PM

We can add the `mb-0` class to [this line](https://github.com/open-craft/frontend-app-course-authoring/pull/44/files#diff-e75dfcdd310f6aa37818e33aeee0c238e2be6d7037a0578cba2b368cf614e5f3R61) in this file in order to fix the issue.

File: src/library-authoring/LibraryAuthoringPage.tsx [hunk 2]
```diff
@@ -82,47 +88,51 @@ const LibraryAuthoringPage = () => {
...
+          </div>
+          <Tabs
+            variant="tabs"
+            activeKey={tabKey}
+            onSelect={handleTabChange}
+            className="my-3"
+          >
+            <Tab eventKey={TAB_LIST.home} title="Home" />
+            <Tab eventKey={TAB_LIST.components} title="Components" />
+            <Tab eventKey={TAB_LIST.collections} title="Collections" />
```
Comment (yusuf-musleh) [inline src/library-authoring/LibraryAuthoringPage.tsx]: nit: The titles should be translated (this should probably be changed in the base branch cc: @ChrisChV)

File: src/library-authoring/components/LibraryComponents.tsx (hunk 1 above)
Comment (yusuf-musleh) [reply src/library-authoring/components/LibraryComponents.tsx:66]: I think it looks much nicer without the loading, nice addition!

**Review by yusuf-musleh — APPROVED** [review]
@rpenido :+1: Great work on this! I've tested it out and everything is looking good, just left a new few nits/comments.

- [X] I tested this:
    - Tested filtering by tags
    - Tested searching for tags to filter by
    - Tested filtering by block type
    - Tested clearing all applied filters
    - Tested clearing a single applied filter
    - Tested searching still works as expected
- [X] I read through the code
- [X] I checked for accessibility issues
- [ ] ~Includes documentation~

File: src/library-authoring/LibraryAuthoringPage.tsx (hunk 2 above)
Comment (rpenido) [reply src/library-authoring/LibraryAuthoringPage.tsx]: Good catch!

I changed it in the base branch 905dbb58de46bc4da9e9565c832b94fcc696563f, and also picked it here c453ef035c6448337e6e04109243173e8a7b8b81

File: src/search-manager/FilterByTags.tsx
Comment (rpenido) [reply src/search-manager/FilterByTags.tsx:1]: Fixed here: 1718c26bf845ea6eb959fb7b005287210913016d

**Comment by rpenido** [comment]
Closed in favor of: https://github.com/openedx/frontend-app-course-authoring/pull/1141

**Pull request closed without merging**